pkg/k8s/k8s_service.go

```diff
@@ -142,3 +144,92 @@ func prepareService(
 	}
 	return svc, nil
 }
+
+func (c *Client) WaitForService(ctx context.Context, name string) error {
+	for {
+		service, err := c.clientset.CoreV1().Services(c.namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return ErrGettingService.WithParams(name).Wrap(err)
+		}
+
+		if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+			if len(service.Status.LoadBalancer.Ingress) == 0 {
+				time.Sleep(waitRetry)
+				continue // Wait until the LoadBalancer IP is available
+			}
+		} else if service.Spec.Type == v1.ServiceTypeNodePort {
+			if service.Spec.Ports[0].NodePort == 0 {
+				return ErrNodePortNotSet
```

Comment:
Same comment on why returning an error and not continuing.

Reply:
Here if there is an error on getting the service like some internal k8s server error or communication error, we would want to propagate it back to the user. The service must be there and it should be queriable because when user calls this function is when a service is created and want to know when it is ready to be used.